lasy/profiles/speckle_profile.py

```diff
@@ -0,0 +1,371 @@
+
+        # # ================== Calculate auxiliary variables ================== #
+        if "SSD" in self.lsType.upper():
+            phase_plate = np.random.uniform(
+                -np.pi, np.pi, size=self.n_beamlets[0] * self.n_beamlets[1]
+            ).reshape(self.n_beamlets)
+        elif "ISI" in self.lsType.upper():
+            phase_plate = np.zeros(self.n_beamlets)  # ISI does not require phase plates
+        else:
+            raise NotImplementedError
+
+        ssd_frac = np.sqrt(self.ssd_distr[0] ** 2 + self.ssd_distr[1] ** 2)
+        ssd_frac = self.ssd_distr[0] / ssd_frac, self.ssd_distr[1] / ssd_frac
+        phase_mod_freq = [
+            self.laser_bandwidth * sf * 0.5 / pma
+            for sf, pma in zip(ssd_frac, self.phase_mod_amp)
+        ]
+        x_lens_list = np.linspace(
+            -0.5 * (self.n_beamlets[0] - 1),
+            0.5 * (self.n_beamlets[0] - 1),
+            num=self.n_beamlets[0],
+        )
+        y_lens_list = np.linspace(
+            -0.5 * (self.n_beamlets[1] - 1),
+            0.5 * (self.n_beamlets[1] - 1),
+            num=self.n_beamlets[1],
+        )
+        Y_lens_matrix, X_lens_matrix = np.meshgrid(y_lens_list, x_lens_list)
+        Y_lens_index_matrix, X_lens_index_matrix = np.meshgrid(
+            np.arange(self.n_beamlets[1], dtype=float),
+            np.arange(self.n_beamlets[0], dtype=float),
+        )
+        phase_mod_phase = np.random.standard_normal(2) * np.pi
+        td = (
+            self.ncc[0] / phase_mod_freq[0] if phase_mod_freq[0] > 0 else 0,
+            self.ncc[1] / phase_mod_freq[1] if phase_mod_freq[1] > 0 else 0,
+        )
+        stochastic_process_time = np.arange(0, tmax + self.tu, self.tu)
+
+        # ======================= Initialization ========================= #
+        def gen_gaussian_time_series(t_num, fwhm, rms_mean):
+            """Generate a discrete time series that has gaussian power spectrum.
+
+            :param t_num: number of grid points in time
+            :param fwhm: full width half maximum of the power spectrum
+            :param rms_mean: root-mean-square average of the spectrum
+            :return: a time series array of complex numbers with shape [t_num]
+            """
+            if fwhm == 0.0:
+                return np.zeros((2, t_num))
+            else:
+                omega = np.fft.fftshift(np.fft.fftfreq(t_num, d=self.tu))
+                # rand_ph = np.random.normal(scale=np.pi, size=t_num)
+                psd = np.exp(-np.log(2) * 0.5 * np.square(omega / fwhm * 2 * np.pi))
+                psd *= np.sqrt(t_num) / np.sqrt(np.mean(np.square(psd))) * rms_mean
+                pm_phase = np.array(psd) * (
+                    np.random.normal(size=t_num) + 1j * np.random.normal(size=t_num)
+                )
+                pm_phase = np.fft.ifftshift(np.fft.fft(np.fft.fftshift(pm_phase)))
+                pm_phase *= rms_mean / np.sqrt(np.mean(np.square(np.abs(pm_phase))))
```

Comment:
Maybe we could use more explicit names
```suggestion
                spectral_amplitude = np.array(psd) * (
                    np.random.normal(size=t_num) + 1j * np.random.normal(size=t_num)
                )
                temporal_amplitude = np.fft.ifftshift(np.fft.fft(np.fft.fftshift(spectral_amplitude)))
                temporal_amplitude *= rms_mean / np.sqrt(np.mean(np.square(np.abs(temporal_amplitude))))
```